mesonbuild/mtest.py

```diff
@@ -1606,9 +1606,12 @@ def load_metadata(self) -> None:
             # happen before rebuild_deps(), because we need the correct list of
             # tests and their dependencies to compute
             if not self.options.no_rebuild:
-                ret = subprocess.run(self.ninja + ['build.ninja']).returncode
-                if ret != 0:
-                    raise TestException(f'Could not configure {self.options.wd!r}')
+                teststdo = subprocess.run(self.ninja + ['-n', 'build.ninja'], capture_output=True, text=True).stdout
+                if 'ninja: no work to do.' not in teststdo and 'samu: nothing to do' not in teststdo:
+                    stdo = sys.stderr if self.options.list else sys.stdout
```

Comment:
Maybe we can avoid double invocation by always redirecting ninja to stderr? I think that's what @anarazel suggested?

Reply:
But then we always get "ninja: no work to do" printed before any tests are run, which is kind of ugly.

From the original commit message by @anarazel

```
    One issue with this is that that we will now output more useless ninja output
    when nothing needs to be done (the "Entering directory" part is not repeated,
    as we happen to be in the build directory already). It likely is worth
    removing that output, perhaps by testing if anything needs to be done with
    ninja -n, but that seems better addressed separately.
```